lib/questionnaires/model/item/src/question_item_model.dart

```diff
@@ -316,9 +316,17 @@ class QuestionItemModel extends ResponseItemModel {
       firstAnswerModel.populateFromExpression(initialEvaluationResult);
     } else {
       // initial.value[x]
-      final initialValues = questionnaireItem.initial;
+      final initialValues = questionnaireItem.initial ?? [];
+
+      if ({QuestionnaireItemType.choice,QuestionnaireItemType.open_choice}.contains(questionnaireItem.type)) {
+        // Add answerOptions marked as initialSelected to array of initialValues
+        final initialSelected = questionnaireItem.answerOption
+            ?.where((option) => option.initialSelected?.value == true && option.valueCoding != null)
+            .map((option) => QuestionnaireInitial(valueCoding: option.valueCoding));
+        initialValues.addAll(initialSelected ?? []);
+      }
```

Comment:
Just want to confirm, since we are using `addAll()`, so it's possible `initialValues` has values from both `questionnaireItem.initial` + `questionnaireItem.answerOption`. Is that correct?

Reply:
That is correct, in theory. 

The original implementation uses `Questionnaire.item.initial` to set up the initial values for every single question type. Here we are basically appending all `initialSelected` options to the same `initialValues` array, so if a question has both `answerOptions` with `initialSelected` and values in the `initial` array, these may get mixed.

That said, according to the FHIR Questionnaire spec (http://hl7.org/fhir/R4/questionnaire-definitions.html#Questionnaire.item) you cannot have a question with both `answerOptions` and `initial` at the same time, so as long as the FHIR Questionnaire is strictly validated, this shouldn't happen in theory.

Because of this, I was actually considering dropping support for `initial` in `choice` and `open-choice`, since these always have an `answerOptions` array, but decided to keep things as they are for now to not break any Questionnaires that may have potentially been made in this way already.

Reply:
> That said, according to the FHIR Questionnaire spec (http://hl7.org/fhir/R4/questionnaire-definitions.html#Questionnaire.item) you cannot have a question with both answerOptions and initial at the same time, so as long as the FHIR Questionnaire is strictly validated, this shouldn't happen in theory.

In that case, we can proceed with the current implementation, thank you so much for the explanation!